bin/stashcp

```diff
@@ -265,6 +271,110 @@ def doStashCpSingle(sourceFile, destination, debug=False):
             return 1
     return 0
 
+def check_for_xrootd():
+    """
+    Check if xrootd is installed by checking if the xrdcp command returns a reasonable output
+    """
+    # xrdcp output the version on stderr, what?!?
+    check_command = "xrdcp -V 2>&1"
+    logging.debug("Running the command to check of xrdcp existance: %s", check_command)
+    command_object = subprocess.Popen([check_command], stdout=subprocess.PIPE, shell=True)
+    xrdcp_version = command_object.communicate()[0]
+    if command_object.returncode == 0:
+        logging.debug("xrdcp version: %s", xrdcp_version)
+        return xrdcp_version
+    else:
+        logging.debug("xrdcp command returned exit code: %i", command_object.returncode)
+        return False
+
+
+def download_with_http(source, destination, debug):
+    """
+    Download from the nearest cache with HTTP
+    """
+    global nearest_cache
+    global nearest_cache_list
+    logging.debug("Downloading with HTTP")
+
+    payload = {}
+    payload['filename'] = source
+    sitename = os.environ.setdefault("OSG_SITE_NAME", "siteNotFound")
+    payload['sitename'] = sitename
+    payload.update(parse_job_ad())
+
+    if not nearest_cache:
+        nearest_cache = get_best_stashcache()
+
+    # Ok, now run the curl command:
+    if debug:
+        output_mode = "-v"
+    else:
+        output_mode = "-s"
+
+    # The command will cd into destination directory and then run curl
+    if os.path.isdir(destination):
+        destination += "/"
+    dest_dir, dest_filename = os.path.split(destination)
+    if not dest_dir:
+        dest_dir = "."
+
+    if dest_filename:
+        download_output = "-o %s" % dest_filename
+        final_destination = destination
+    else:
+        download_output = "-O"
+        final_destination = os.path.join(dest_dir, os.path.basename(source))
+    
+    # Try 2 nearest caches
+    success = False
+    start = end = 0
+    tried_cache = ""
+    for cache in nearest_cache_list[:2]:
+        tried_cache = cache
+        # Parse the nearest_cache url, make sure it uses http
+        # Should really use urlparse, but python3 and python2 urlparse imports are 
+        # very different
+        if cache.startswith('root://'):
+            cache = cache.replace('root://', 'http://')
+
+        # Append port 8000, which is just a convention for now, not set in stone
+        cache += ":8000"
+        curl_command = "cd %s; curl %s -L --connect-timeout 30 --speed-limit 1024 %s --fail %s%s" % (dest_dir, output_mode, download_output, cache, source)
```

Comment:
This is totally gonna break if `dest_dir` has a space in it. I suggest quoting it -- either with `re.quote()` or `pipes.quote()`.